datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -849,12 +849,61 @@ impl<'a, S: SimplifyInfo> ExprRewriter for Simplifier<'a, S> {
                 out_expr.rewrite(self)?
             }
 
+            // concat
+            ScalarFunction {
+                fun: BuiltinScalarFunction::Concat,
+                args,
+            } => {
+                let mut new_args = Vec::with_capacity(args.len());
+                let mut contiguous_scalar = "".to_string();
+                for e in args {
+                    match e {
+                        // ignore `null` scalar and concatenate it with `contiguous scalar`.
+                        Expr::Literal(x) => {
+                            match x {
+                                // true --> '1', false --> '0'
+                                ScalarValue::Boolean(b) => {
+                                    contiguous_scalar += b
+                                        .map(|b| if b { "1" } else { "0" })
```

Comment:
The behavior of `to_string` doesn't 100% match the `cast_to_utf8` kernel in `arrow-rs`. 
(BooleanType -> '1' / '0', and `null -> empty string)
Maybe we should reimplement it or add a `cast_to_utf8` function for `ScalarValue`.

Reply:
I agree finding some way to make the behavior the same is important if datafusion will rewrite the expression

I think in general, we should not try to reimplement the behavior in the simplify logic. By the time an expresison gets here, it should have been cast to `Utf8` anyways -- by the type coercion logic. 

I would recommend only simplifying calls where all the arguments are ScalarValue::Utf8